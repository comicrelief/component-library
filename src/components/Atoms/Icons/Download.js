import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';

const StyledSVG = styled.svg`
  // Mobile-colour if available, else use standard prop
  fill: ${({ mobileColour, colour, theme }) => (mobileColour ? theme.color(mobileColour) : theme.color(colour))};

  // Reinstate standard styles for 'desktop', adding a fallback for good measure:
  @media ${({ theme }) => theme.allBreakpoints('L')} {
    fill: ${({ colour, theme }) => (colour ? theme.color(colour) : theme.color('white'))};
  }
`;

const Download = ({
  colour = 'white', mobileColour = null, theme, size = 24, ...rest
}) => (
  <StyledSVG
    {...rest}
    width={size}
    height={size}
    fill={theme.color(colour)}
    colour={colour}
    mobileColour={mobileColour}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 96 96"
  >
    <path d="M17.171 38.829a4 4 0 015.657-5.657L44 54.343V8a4 4 0 018 0v46.343l21.172-21.172a4 4 0 115.656 5.657l-28 28-.005.004a3.976 3.976 0 01-.604.493c-.101.068-.211.115-.317.172-.123.067-.241.141-.372.195-.133.055-.273.087-.41.127-.113.033-.22.077-.337.1a3.89 3.89 0 01-1.566.002c-.116-.023-.224-.067-.336-.1-.138-.04-.277-.072-.411-.128-.13-.054-.248-.128-.37-.195-.106-.058-.216-.105-.318-.173a4.037 4.037 0 01-.604-.493l-.005-.004-28.002-27.999zM84 68a4 4 0 00-4 4v12H16V72a4 4 0 00-8 0v16a4 4 0 004 4h72a4 4 0 004-4V72a4 4 0 00-4-4z" />
  </StyledSVG>
);

Download.propTypes = {
  colour: PropTypes.string,
  mobileColour: PropTypes.string,
  size: PropTypes.number,
  theme: PropTypes.objectOf(PropTypes.shape).isRequired
};

export default withTheme(Download);
