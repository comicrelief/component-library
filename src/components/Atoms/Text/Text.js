import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

/**
 * Text component
 */
export const BaseText = styled.span`
  color: ${({ color, theme }) =>
    color ? theme.color(color) : 'inherit'};
  font-size: ${({ size, theme }) => theme.fontSize(size)};
`;

/** Text renders different elements based on the `tag` prop */
const Text = ({ tag, children, ...rest }) => (
  <BaseText {...rest} as={tag}>
    {children}
  </BaseText>
);

Text.defaultProps = {
  tag: 'span',
  weight: 'normal',
  uppercase: false,
  size: 's0'
};

Text.propTypes = {
  /** Weight of Font */
  weight: PropTypes.string,
  /** Sets text transform to uppercase. */
  uppercase: PropTypes.bool,
  /** Colors */
  color: PropTypes.string,
  /** Tag type */
  tag: PropTypes.string,
  /** Sizes */
  size: PropTypes.string
};

export default Text;
