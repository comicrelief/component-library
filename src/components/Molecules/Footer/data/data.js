export default {
  menuDescription: 'Footer',
  menuGroup: [
    {
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      title: 'Hear from us',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Get the newsletter',
          url: 'https://www.comicrelief.com/#',
          class: null
        },
        {
          title: 'How we contact you',
          url: 'https://www.comicrelief.com/update-your-preferences',
          class: null
        }
      ]
    },
    {
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      title: 'Get in touch',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Contact us',
          url: 'https://www.comicrelief.com/contact-us',
          class: null
        },
        {
          title: 'Your Gift Aid',
          url: 'https://giftaid.comicrelief.com/update',
          class: null
        },
        {
          title: 'FAQs',
          url: 'https://www.comicrelief.com/frequently-asked-questions',
          class: null
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          url:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      title: 'About us',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Mission',
          url: 'https://www.comicrelief.com/about-comic-relief',
          class: null
        },
        {
          title: 'Our history',
          url: 'https://www.comicrelief.com/about-comic-relief/history',
          class: null
        },
        {
          title: 'Finances',
          url: 'https://www.comicrelief.com/about-comic-relief/finances',
          class: null
        },
        {
          title: 'Partners',
          url: 'https://www.comicrelief.com/working-with-us',
          class: null
        },
        {
          title: 'Meet the team',
          url: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          class: null
        },
        {
          title: 'Sport Relief',
          url: 'https://www.sportrelief.com/',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      title: 'Careers',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Working at Comic Relief',
          url: 'https://www.comicrelief.com/careers',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://app.beapplied.com/org/comic-relief',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      title: 'News',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'News',
          url: 'https://www.comicrelief.com/news',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://www.comicrelief.com/press-releases',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      title: 'Legal',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Privacy policy',
          url: 'https://www.comicrelief.com/privacy-notice',
          class: null
        },
        {
          title: 'Terms of use',
          url: 'https://www.comicrelief.com/terms-of-use',
          class: null
        },
        {
          title: 'Cookies',
          url: 'https://www.comicrelief.com/cookies-policy',
          class: null
        },
        {
          title: 'Positive practices',
          url: 'https://www.comicrelief.com/positive-practices',
          class: null
        },
        {
          title: 'Text to donate terms of use',
          url: 'https://www.comicrelief.com/legal/sms',
          class: null
        }
      ]
    }
  ]
};
