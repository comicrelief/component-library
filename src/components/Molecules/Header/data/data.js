export default {
  menuDescription: 'I am the navigation',
  menuGroup: [
    {
      id: 'Parent Item Title 1',
      title: 'Parent Item Title 1',
      url: '/parent1',
      links: [
        {
          title: 'Child Item Title 1-1',
          url: '/child-item-1-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-2',
          url: '/child-item-1-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-3',
          url: '/child-item-1-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 2',
      title: 'Parent Item Title 2',
      url: '/parent2',
      links: [
        {
          title: 'Child Item Title 2-1',
          url: '/child-item-2-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-2',
          url: '/child-item-2-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-3',
          url: '/child-item-2-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 4',
      title: 'Parent Item Title 4',
      url: '/parent4',
      links: [
        {
          title: 'Child Item Title 4-1',
          url: '/child-item-4-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-4',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    }
  ]
};
