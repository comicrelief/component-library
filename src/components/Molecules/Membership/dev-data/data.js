export default {
  cartId: 'test-sept2019-C365',
  regularGiving: {
    otherDescription:
      'Your donation will help us fund amazing projects in the UK and around the world.',
    moneybuys: [
      {
        amount: 10,
        description:
          'a regular supply of toiletries for someone living in a refugee camp in Serbia.'
      },
      {
        amount: 20,
        description:
          'a potentially lifesaving call for a man at risk of suicide in the UK.'
      },
      {
        amount: 30,
        description:
          'the distribution of enough surplus food for a school breakfast club to feed 80 children.'
      }
    ]
  }
};
