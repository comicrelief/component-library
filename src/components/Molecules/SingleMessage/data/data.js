export default {
  title: 'What your money does',
  text:
    'When you donate to Comic Relief or Sport Relief, youâ€™re supporting vulnerable people and communities in the UK and internationally.',
  pictures: {
    small: 'src/components/Atoms/Picture/assets/picture-1.jpg',
    medium: 'src/components/Atoms/Picture/assets/picture-2.jpg',
    large: 'src/components/Atoms/Picture/assets/picture-3.jpg'
  },
  ctaText: 'Find out how'
};
