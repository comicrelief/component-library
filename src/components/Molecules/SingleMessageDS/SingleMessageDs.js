import React from 'react';
import PropTypes from 'prop-types';

import Picture from '../../Atoms/Picture/Picture';
import Link from '../../Atoms/Link/Link';
import { External, Internal } from '../../Atoms/Icons/index';
import {
  Container, Copy, CTA, MediaLink, Image, Subtitle
} from './SingleMessageDs.style';

const SingleMessageDs = ({
  backgroundColor,
  imageLeft,
  imageLow,
  images,
  image,
  imageAltText,
  height,
  width,
  subtitle,
  children,
  link,
  linkLabel,
  ctaBgColor,
  target,
  ...rest
}) => {
  const Media = (
    <Image>
      <Picture
        alt={imageAltText}
        imageLow={imageLow}
        images={images}
        image={image}
        objectFit="cover"
        width={width}
        height={height}
      />
    </Image>
  );

  const hasMedia = () => {
    if (imageLow && link) {
      return (
        <MediaLink
          aria-hidden="true"
          tabIndex="-1"
          href={link}
          target={target}
          {...rest}
        >
          {Media}
        </MediaLink>
      );
    }
    if (imageLow) {
      return Media;
    }
    return null;
  };

  const icon = () => {
    if (target === 'blank') {
      return <External colour="white" />;
    }
    return <Internal colour="white" />;
  };

  const external = target === 'blank' ? 'noopener noreferrer' : null;
  const subtitleColour = ctaBgColor === 'red' ? 'grey_dark' : ctaBgColor;

  return (
    <Container {...rest} imageLeft={imageLeft}>
      {hasMedia()}
      <Copy
        hasImage={imageLow}
        hasLink={link}
        backgroundColor={backgroundColor}
        imageLeft={imageLeft}
      >
        <Subtitle color={subtitleColour} size="s" weight="bold" family="Montserrat">
          {subtitle}
        </Subtitle>
        {children}
        {link && (
        <CTA hasImage={imageLow}>
          <Link
            rel={external}
            color={ctaBgColor}
            href={link}
            target={target}
            type="button"
            icon={icon()}
          >
            {linkLabel}
          </Link>
        </CTA>
        )}
      </Copy>
    </Container>
  );
};

SingleMessageDs.propTypes = {
  backgroundColor: PropTypes.string,
  imageLeft: PropTypes.bool,
  imageLow: PropTypes.string,
  images: PropTypes.string,
  image: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  imageAltText: PropTypes.string,
  link: PropTypes.string,
  ctaBgColor: PropTypes.string,
  linkLabel: PropTypes.string,
  subtitle: PropTypes.string.isRequired,
  target: PropTypes.string,
  children: PropTypes.node.isRequired
};

SingleMessageDs.defaultProps = {
  backgroundColor: 'white',
  imageLeft: true,
  imageLow: null,
  images: null,
  image: null,
  link: null,
  ctaBgColor: 'red',
  linkLabel: null,
  target: null,
  imageAltText: '',
  width: '100%',
  height: '100%'
};

export default SingleMessageDs;
