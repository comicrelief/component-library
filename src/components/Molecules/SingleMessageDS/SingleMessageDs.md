# SingleMessageDs

## Image align left and CTA are clickable and links user through to full conten

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  letter-spacing: 0.03em;
  text-transform: uppercase;
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      imageLeft={true}
      link='/home'
      linkLabel='Check out'
      backgroundColor='white'
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
      hasVideo={true}
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Image align right  and CTA are clickable and links user through to full content

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      target='blank'
      link='https://www.comicrelief.com/'
      linkLabel='Check out'
      ctaBgColor="blue_dark"
      backgroundColor='white'
      imageLeft={false}
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Image align with non CTA

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Title = styled(Text)`
  margin: ${spacing('md')} 0;
  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('m')};
  }
`;
<div
    style={{
      display: 'flex',
      padding: '4rem 0',
      background: '#E1E2E3',
      justifyContent: 'space-around',
    }}>
    <SingleMessageDs
      linkLabel='Check out'
      ctaBgColor="blue_dark"
      backgroundColor='white'
      imageLeft={true}
      imageLow={data.image}
      images={data.images}
      imageAltText='Happy man going to work'
      subtitle="Subtitle"
      height='100%'
    >
      <Title tag="h4" color="grey_dark" size="xl" family="Anton" weight="normal">
        Heading Line 1 Heading Line 2
      </Title>
      <Text tag='p'>
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
</div>;
```

## Using the linkIcon prop (to display download icon)

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import styled from 'styled-components';
import spacing from '../../../theme/shared/spacing';
import Download from '../../Atoms/Icons/Download';

<div>
    <SingleMessageDs
      link="https://assets.ctfassets.net/zsfivwzfgl3t/EsIEuI1teYkbLB7peOtqP/b43a905cffd89115d642fe5a84ff020a/RND21_fundraise_moneybuys_poster.pdf"
      linkLabel='Download this'
      linkIcon={<Download />}
      imageLow={data.image}
      images={data.images}
    >
      <Text tag='p'>
        Some text and stuff.
      </Text>
    </SingleMessageDs>
</div>;
```
