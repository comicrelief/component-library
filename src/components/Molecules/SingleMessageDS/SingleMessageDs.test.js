import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import SingleMessageDs from './SingleMessageDs';
import Text from '../../Atoms/Text/Text';
import Download from '../../Atoms/Icons/Download';
import { defaultData } from '../../../styleguide/data/data';
it('renders correctly', () => {
  const tree = renderWithTheme(
    <SingleMessageDs
      target="_blank"
      link="/home"
      linkLabel="Check out the shop"
      ctaBgColor="blue_dark"
      backgroundColor="white"
      imageDirection="right"
      imageLow={defaultData.image}
      images={defaultData.images}
      imageAltText="Happy man going to work"
      subtitle="Subtitle"
      title="Heading Line 1 Heading Line 2"
      height="auto"
      imageLeft
      linkIcon={<Download />}
    >
      <Text tag="p">
        Whatever you’ve got planned, the Sport Relief shop has everything you need
        to get you looking your best while you’re raising some cash. Also
        available in Sainsbury’s stores and online and in selected Argos stores.
      </Text>
    </SingleMessageDs>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
