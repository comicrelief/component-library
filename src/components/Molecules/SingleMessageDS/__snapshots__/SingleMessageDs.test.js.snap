// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c6 {
  color: #274084;
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  font-weight: bold;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c8 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c3 {
  display: block;
  width: 100%;
  height: auto;
  position: relative;
}

.c4 {
  width: 100%;
  height: auto;
  display: block;
  object-fit: cover;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: 1rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  right: 1.5rem;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  padding: 0.5rem 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 1rem;
  border-radius: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  height: 3.125rem;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #2042AD;
  color: #FFFFFF;
}

.c11:hover {
  background-color: #274084;
  color: #FFFFFF;
}

.c13 {
  fill: #FFFFFF;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  background: inherit;
}

.c7 {
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  text-transform: uppercase;
}

.c2 {
  height: 100%;
}

.c2 img {
  border-radius: 1rem;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.15);
}

.c1 {
  margin-right: 1.5rem;
}

.c10 {
  position: absolute;
  bottom: -1.5rem;
  z-index: 2;
}

.c10 a {
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c5 {
  position: relative;
  height: auto;
  padding: 1.5rem;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 1rem;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.15);
  background: #FFFFFF;
  padding-bottom: calc(2rem + 1.5rem);
  margin: -1.5rem 0 1.5rem 1.5rem;
}

.c5 .c9 {
  right: 1.5rem;
}

@supports not (object-fit:cover) {
  .c3 {
    background-color: yellow !important;
    background-image: url(//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg);
    background-size: cover;
    background-position: center;
  }
}

@supports not (object-fit:cover) {
  .c4 {
    visibility: hidden;
  }
}

@media (min-width:740px) {
  .c12 {
    width: auto;
    right: auto;
    position: relative;
  }
}

@media (min-width:740px) {
  .c11 {
    width: auto;
  }
}

@media (min-width:1024px) {
  .c11 {
    background-color: #2042AD;
    color: #FFFFFF;
  }

  .c11:hover {
    background-color: #274084;
    color: #FFFFFF;
  }
}

@media (min-width:1024px) {
  .c13 {
    fill: #FFFFFF;
  }
}

@media (min-width:740px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c2 {
    height: 448px;
  }
}

@media (min-width:740px) {
  .c1 {
    margin: 0;
    height: 448px;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media (min-width:740px) {

}

@media (min-width:740px) {
  .c5 {
    min-height: 448px;
    height: auto;
    width: calc(50% + 1.5rem);
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    -webkit-flex-basis: calc(50% + 1.5rem);
    -ms-flex-preferred-size: calc(50% + 1.5rem);
    flex-basis: calc(50% + 1.5rem);
    padding: 2rem;
    padding-bottom: calc(2rem + 1.5rem);
    margin: 2rem 0 1.5rem -1.5rem;
  }
}

@media (min-width:1440px) {
  .c5 {
    padding: calc(1.5rem * 2);
    padding-bottom: calc(1.5rem * 3);
    margin: 2rem 0 1.5rem -2rem;
  }
}

<div
  className="c0"
  title="Heading Line 1 Heading Line 2"
>
  <a
    aria-hidden="true"
    className="c1"
    data-image-cta-copy="Check out the shop"
    href="/home"
    tabIndex="-1"
    target="_blank"
    title="Heading Line 1 Heading Line 2"
  >
    <div
      className="c2"
    >
      <div
        className="c3 lazyload"
        height="auto"
        width="100%"
      >
        <img
          alt="Happy man going to work"
          className="c4 lazyload"
          data-lowsrc="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          data-sizes="auto"
          data-src={null}
          data-srcset="//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=200&h=150&q=50 200w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=400&h=300&q=50 400w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=800&h=600&q=50 800w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=1200&h=900&q=50 1200w,//images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg?w=1440&h=1080&q=50 1440w"
          height="auto"
          src={null}
          srcSet="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          width="100%"
        />
      </div>
    </div>
  </a>
  <div
    className="c5"
  >
    <span
      className="c6 c7"
      color="blue_dark"
      size="s"
    >
      Subtitle
    </span>
    <p
      className="c8"
      color="inherit"
      size="s"
    >
      Whatever you’ve got planned, the Sport Relief shop has everything you need to get you looking your best while you’re raising some cash. Also available in Sainsbury’s stores and online and in selected Argos stores.
    </p>
    <div
      className="c9 c10"
    >
      <a
        className="c11"
        color="blue_dark"
        data-button-cta-copy="Check out the shop"
        href="/home"
        rel={null}
        target="_self"
        type="button"
      >
        Check out the shop
        <span
          className="c12"
          type="button"
        >
          <svg
            className="c13"
            fill="#FFFFFF"
            height={24}
            viewBox="0 0 96 96"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.171 38.829a4 4 0 015.657-5.657L44 54.343V8a4 4 0 018 0v46.343l21.172-21.172a4 4 0 115.656 5.657l-28 28-.005.004a3.976 3.976 0 01-.604.493c-.101.068-.211.115-.317.172-.123.067-.241.141-.372.195-.133.055-.273.087-.41.127-.113.033-.22.077-.337.1a3.89 3.89 0 01-1.566.002c-.116-.023-.224-.067-.336-.1-.138-.04-.277-.072-.411-.128-.13-.054-.248-.128-.37-.195-.106-.058-.216-.105-.318-.173a4.037 4.037 0 01-.604-.493l-.005-.004-28.002-27.999zM84 68a4 4 0 00-4 4v12H16V72a4 4 0 00-8 0v16a4 4 0 004 4h72a4 4 0 004-4V72a4 4 0 00-4-4z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;
